tests/test_result.py

```diff
@@ -0,0 +1,157 @@
+    obj = heracles.Result(arr)
+    np.testing.assert_array_equal(obj, arr)
+    assert type(obj) is heracles.Result
+    assert obj.axis == 0
+    assert obj.ell is None
+    assert obj.lower is None
+    assert obj.upper is None
+    assert obj.weight is None
+
+    sliced = obj[1:]
+    np.testing.assert_array_equal(sliced, arr[1:])
+    assert type(sliced) is heracles.Result
+    assert sliced.axis == 0
+    assert sliced.ell is None
+    assert sliced.lower is None
+    assert sliced.upper is None
+    assert sliced.weight is None
+
+    ell = np.arange(lmax + 1)
+    ellmin = ell
+    ellmax = ell + 1
+    weight = rng.random(lmax + 1)
+    obj = heracles.Result(arr, ell, lower=ellmin, upper=ellmax, weight=weight)
+    np.testing.assert_array_equal(obj, arr)
+    assert type(obj) is heracles.Result
+    assert obj.axis == 0
+    assert obj.ell is ell
+    assert obj.lower is ellmin
+    assert obj.upper is ellmax
+    assert obj.weight is weight
+
+    sliced = obj[1:]
+    np.testing.assert_array_equal(sliced, arr[1:])
+    assert type(sliced) is heracles.Result
+    assert sliced.axis == 0
+    assert sliced.ell is ell
+    assert sliced.lower is ellmin
+    assert sliced.upper is ellmax
+    assert sliced.weight is weight
+
+    arr = rng.random((lmax + 1, 100))
+    obj = heracles.Result(arr, ell, lower=ellmin, upper=ellmax, weight=weight, axis=0)
+    np.testing.assert_array_equal(obj, arr)
+    assert type(obj) is heracles.Result
+    assert obj.ell is ell
+    assert obj.lower is ellmin
+    assert obj.upper is ellmax
+    assert obj.weight is weight
+    assert obj.axis == 0
+
+    copy = obj.copy()
+    copy[:] += 1.0
+    np.testing.assert_array_equal(copy, arr + 1.0)
+    assert type(copy) is heracles.Result
+    assert copy.ell is ell
+    assert copy.lower is ellmin
+    assert copy.upper is ellmax
+    assert copy.weight is weight
+    assert copy.axis == 0
+
+    view = obj.view(heracles.Result)
+    np.testing.assert_array_equal(view, arr)
+    assert type(view) is heracles.Result
+    assert view.ell is ell
+    assert view.lower is ellmin
+    assert view.upper is ellmax
+    assert view.weight is weight
+    assert view.axis == 0
+
+    with pytest.raises(ValueError, match="axis 1 is out of bounds"):
+        heracles.Result([], axis=1)
+
+
+@pytest.mark.parametrize("weight", [None, "l(l+1)", "2l+1", "<rand>"])
+@pytest.mark.parametrize("ndim,axis", [(1, 0), (2, 0), (3, 1)])
+def test_binned(ndim, axis, weight, rng):
+    shape = rng.integers(0, 100, ndim)
+    lmax = shape[axis] - 1
+
+    data = heracles.Result(rng.standard_normal(shape), axis=axis)
+
+    nbins = rng.integers(1, 10)
+    bins = rng.integers(1, lmax + 1, nbins, endpoint=True)
+    bins.sort()
+
+    if weight == "<rand>":
+        weight = rng.random(lmax + 1)
+
+    result = heracles.binned(data, bins, weight)
+
+    ell = np.arange(lmax + 1)
+
+    if weight is None:
+        w = np.ones_like(ell)
+    elif isinstance(weight, str):
```

Comment:
Again I find this string for weights very elegant

Reply:
*Not* very elegant, presumably?